pylint/checkers/similar.py

```diff
@@ -369,11 +371,11 @@ def __init__(
         self.linesets: List["LineSet"] = []
 
     def append_stream(
-        self, streamid: str, stream: TextIOWrapper, encoding=None
+        self, streamid: str, stream: Union[BufferedReader, BytesIO], encoding=None
     ) -> None:
         """append a file to search for similarities"""
         if encoding is None:
-            readlines = stream.readlines
+            readlines: Union[Callable[..., Union[str, bytes]], Any] = stream.readlines
```

Comment:
We might want to refactor this, not the best of typing..

Reply:
I agree, but maybe in another merge request ?